### PR TITLE
Drop modifiers

### PR DESCRIPTION
**Pull request by odersky**
The end goal of this PR is to get rid of the Modifiers structure and to represent each modifier as a tree of its own. This is more IDE friendly and simplifies treatment of trees. 

@DarkDimius can you comment on this? Also, have a look at my comments in BackendInterface which require action.

### CODE REVIEW

File: src/dotty/tools/backend/jvm/DottyBackendInterface.scala [hunk 1]
```diff
@@ -944,7 +944,7 @@ class DottyBackendInterface(outputDirectory: AbstractFile, val superCallsMap: Ma
   }
 
   object ValDef extends ValDefDeconstructor {
-    def _1: Modifiers = field.mods
+    def _1: Modifiers = tpd.Modifiers(field.symbol)
```
Comment (odersky) [inline src/dotty/tools/backend/jvm/DottyBackendInterface.scala]: Note that modifiers will be re-created as a new object everytime mods is called. Is it possible to split accessor into two accessors, one for flags and another for annotations? If not, an alternative would be to let Symbol inherit from Modifiers.

File: src/dotty/tools/dotc/ast/TreeInfo.scala [hunk 2]
```diff
@@ -249,17 +249,6 @@ trait TreeInfo[T >: Untyped <: Type] { self: Trees.Instance[T] =>
   /** Is this case guarded? */
   def isGuardedCase(cdef: CaseDef) = cdef.guard ne EmptyTree
 
-  /** True iff definition is a val or def with no right-hand-side, or it
-   *  is an abstract typoe declaration
```
Comment (odersky) [inline src/dotty/tools/dotc/ast/TreeInfo.scala]: This function was unused, superseded by NavigateAST.

File: src/dotty/tools/backend/jvm/DottyBackendInterface.scala (hunk 1 above)
Comment (xeno-by) [inline src/dotty/tools/backend/jvm/DottyBackendInterface.scala]: Do you also plan to retain original positions of modifiers as written in the source file?

Comment (odersky) [inline src/dotty/tools/backend/jvm/DottyBackendInterface.scala]: In the parse-trees yes. But these are not preserved in typed trees or TASTY.

Comment (DarkDimius) [inline src/dotty/tools/backend/jvm/DottyBackendInterface.scala]: Grepping the source code it seems that _1 is never used. Try returning `null` to ensure it?

**Comment by felixmulder** [comment]
I tried this locally and everything seems to be fine. I'll try to rebuild this again.

**Comment by DarkDimius** [comment]
LGTM

**Pull request merged**